weather/utils.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+import os
+from typing import Callable
+import json
+
+from fastapi import HTTPException
+
+
+def check_write_access():
+    filepath = os.getcwd()
+
+    if os.access(filepath, os.W_OK):
+        print("Access granted")
+    else:
+        print("Denied")
+
+
+def exception_handler(func: Callable) -> Callable:
+    async def wrapper(*args, **kwargs):
+        try:
+            return await func(*args, **kwargs)
+        except Exception as error:
+            check_write_access()
+            with open("exception_log.txt", "a+") as file:
```

Comment:
The `exception_handler` writes exceptions to a file named `exception_log.txt`. Ensure that this file is accessible and writable in the environment where the application runs.